src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToPackAToolTargetingNonSupportedTFM.cs

```diff
@@ -0,0 +1,80 @@
+using Microsoft.NET.TestFramework.Commands;
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.NET.Build.Tasks;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.NET.ToolPack.Tests
+{
+    public class GivenThatWeWantToPackAToolTargetingNonSupportedTFM : SdkTest
+    {
+
+        public GivenThatWeWantToPackAToolTargetingNonSupportedTFM(ITestOutputHelper log) : base(log)
+        {
+
+        }
+
+        [Theory]
+        // lower than netcoreapp2.0
+        [InlineData("TargetFramework", "netcoreapp2.0", "PackAsToolDoesNotSupportTFMLowerThanNetcoreapp21")]
+        [InlineData("TargetFramework", "netcoreapp1.1", "PackAsToolDoesNotSupportTFMLowerThanNetcoreapp21")]
+        [InlineData("TargetFrameworks", "netcoreapp2.0;netcoreapp2.1", "PackAsToolDoesNotSupportTFMLowerThanNetcoreapp21")]
+        // non netcoreapp
+        [InlineData("TargetFramework", "netstandard2.0", "PackAsToolOnlySupportNetcoreapp")]
+        public void It_should_fail_with_error_message(string targetFrameworkProperty, string targetFramework, string error)
+        {
+            TestAsset helloWorldAsset = _testAssetsManager
+                                        .CopyTestAsset("PortableTool", "PackNonSupportedTFM")
+                                        .WithSource()
+                                        .WithProjectChanges(project =>
+                                        {
+                                            XNamespace ns = project.Root.Name.Namespace;
+                                            XElement propertyGroup = project.Root.Elements(ns + "PropertyGroup").First();
+
+                                            propertyGroup.Element(ns + "TargetFramework").Remove();
+                                            propertyGroup.Add(new XElement(ns + targetFrameworkProperty, targetFramework));
+                                        });
+
+            NuGetConfigWriter.Write(helloWorldAsset.TestRoot, new List<string>() { NuGetConfigWriter.DotnetCoreMyGetFeed });
+
+            helloWorldAsset.Restore(Log);
+
+            var packCommand = new PackCommand(Log, helloWorldAsset.TestRoot);
+
+            CommandResult result = packCommand.Execute();
+            result.ExitCode.Should().NotBe(0);
+
+            // walk around attribute requires static
+            string message = null;
+            if (error == "PackAsToolDoesNotSupportTFMLowerThanNetcoreapp21")
+            {
+                message = Strings.PackAsToolDoesNotSupportTFMLowerThanNetcoreapp21;
+            }
+            else if (error == "PackAsToolOnlySupportNetcoreapp")
+            {
+                message = Strings.PackAsToolOnlySupportNetcoreapp;
+            }
+            else
+            {
+                throw new ArgumentException("No matching error to assert");
+            }
+            result.StdOut.Should().Contain(message);
+        }
+
+        [FullMSBuildOnlyFact]
```

Comment:
I think this should be `WindowsOnlyFact` instead of `FullMSBuildOnlyFact`.

Reply:
We don't run our tests on nano though.  There are lots of other tests that assume that full framework is available on Windows.

If we want to support running the tests on Windows without the full framework (and full framework reference assemblies) installed, then we can make our Fact/Theory attributes more granular to represent this, but for now we should get the coverage on the .NET CLI as well as full MSBuild.